aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,15 +3429,36 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if (driver == "gelss" && input.device() != at::kCPU) {
+    if (input.numel() == 0) {
+        auto output_shape = input.sizes().vec();
+        output_shape.back() = other.size(-1);
+        solution.zero_();
+    } else {
+        auto [U, S, Vh] = at::_linalg_svd(input, false, true, "gesvd");
+        rank = at::zeros({1}, at::kLong);
+        rank[0] = (S > rcond).sum();
+        auto S_pinv = S.reciprocal();
+        auto s1 = at::narrow(S, /*dim=*/-1, /*start=*/0, /*length=*/1);  // singular values are sorted in descending order
+        S_pinv.masked_fill_(S < rcond * s1, 0);
+        auto uhOther = at::matmul(U.adjoint(), other);
+        if(S_pinv.dim() != uhOther.dim()) {
+          S_pinv = S_pinv.unsqueeze(-1);
+        }
+        auto S_pinv_other = S_pinv * uhOther;
+        solution = at::matmul(Vh.adjoint(), S_pinv_other);
+    }
+  }
+  else {
+    lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
+  }
   // residuals are available only if m > n and drivers other than gelsy used
   if (m > n && driver != "gelsy") {
     // if the driver is gelss or gelsd then the residuals are available only if rank == n
     bool compute_residuals = true;
     if (driver == "gelss" || driver == "gelsd") {
       if (input.dim() == 2) {
-        compute_residuals = (rank.item().toInt() == n);
+        compute_residuals = (rank.item().toDouble() == n);
```

Comment:
why is this change necessary?

Reply:
Before it would run into integer overflows when the rank wasn't computed. Since we are now computing the rank this can be changed back.